app/src/main/java/com/scouts/kitchenplaner/ui/viewmodel/EditRecipeViewModel.kt

```diff
@@ -0,0 +1,219 @@
+     * @param numberOfPeople New number of people
+     */
+    fun setNumberOfPeople(recipe: Recipe, numberOfPeople: Int) {
+        viewModelScope.launch {
+            editRecipe.setNumberOfPeople(recipe, numberOfPeople)
+        }
+    }
+
+    /**
+     * Adds a dietary speciality to the recipe
+     *
+     * @param recipe The recipe to change
+     * @param speciality The speciality to add
+     */
+    fun addDietarySpeciality(recipe: Recipe, speciality: DietarySpeciality) {
+        viewModelScope.launch {
+            editRecipe.addDietarySpeciality(recipe, speciality)
+        }
+    }
+
+    /**
+     * Deletes a speciality from the recipe
+     *
+     * @param recipe The recipe to change
+     * @param speciality The speciality to delete
+     */
+    fun deleteDietarySpeciality(recipe: Recipe, speciality: DietarySpeciality) {
+        viewModelScope.launch { editRecipe.deleteDietarySpeciality(recipe, speciality) }
+    }
+
+    /**
+     * Adds an ingredient to the given ingredient group, or the whole ingredient group if no ingredient is provided
+     *
+     * @param recipe The recipe to change
+     * @param group The ingredient which should be added, or to which the ingredient is added to
+     * @param ingredient The ingredient to be added ( or null if the ingredient group should be added)
+     */
+    fun addIngredient(recipe: Recipe, group: IngredientGroup, ingredient: Ingredient? = null) {
+        viewModelScope.launch {
+            if (ingredient == null) {
+                editRecipe.addIngredientGroup(recipe, group)
+            } else {
+                editRecipe.addIngredient(recipe, group, ingredient)
+            }
+        }
+
+        /**
+         * Deletes an ingredient group with all its ingredients in it or,
+         * if the ingredient is specified only the ingredient from the group.
+         *
+         * @param recipe The recipe to change
+         * @param group The group to delete or from which the ingredient gets deleted
+         * @param ingredient The ingredient to be deleted ( or null if the ingredient group should be deleted)
+         */
+        fun deleteIngredient(
+            recipe: Recipe, group: IngredientGroup, ingredient: Ingredient? = null
+        ) {
+            viewModelScope.launch {
+                if (ingredient == null) {
+                    editRecipe.deleteIngredientGroup(recipe, group)
+                } else {
+                    editRecipe.deleteIngredient(recipe, group, ingredient)
+                }
+            }
+        }
+
+        /**
+         * Updates an ingredient with the given values. If a value is not specified the old value stays.
+         *
+         * @param recipe The recipe containing the ingredient
+         * @param group The ingredient group the ingredient belongs to
+         * @param ingredient The ingredient to change
+         * @param newName The new name for the ingredient (or null if the old one stays)
```

Comment:
"... (or null if the name should not be changed)"

Similarly for newAmount and newUnit